### PR TITLE
Creates data flow test for multi-column L2-L4

### PR DESCRIPTION
**Pull request by natoromano**
This PR creates a data flow test for multi-column L2-L4 networks, and does a few related fixes in the laminar network API.

@subutai

### CODE REVIEW

File: tests/layers/laminar_network_test.py [hunk 1]
```diff
@@ -63,10 +63,65 @@
...
+    "minThreshold": 10,
+    "predictedSegmentDecrement": 0.004,
+    "activationThreshold": 13,
+    "maxNewSynapseCount": 20,
+    "seed": 42
   },
   "L2Params": {
     "columnCount": 1024,
     "inputWidth": 1024 * 8,
+    "learningMode": 1,
```
Comment (subutai) [inline tests/layers/laminar_network_test.py]: We should probably add the other ColumnPooler parameters to the Region: numActiveColumnsPerInhArea,  synPermProximalInc, synPermProximalDec, initialProximalPermanence.

File: tests/layers/laminar_network_test.py [hunk 2]
```diff
@@ -334,6 +389,319 @@ def testL4L2DataFlow(self):
...
+    # start with an object A input to get L2 representations for object A
+    sensorInput0.addDataToQueue(features0[0], 0, 0)
+    externalInput0.addDataToQueue(locations0[0], 0, 0)
+    sensorInput1.addDataToQueue(features1[0], 0, 0)
+    externalInput1.addDataToQueue(locations1[0], 0, 0)
+    net.run(1)
+
+    # get L2 representation for object B
+    L2RepresentationA0 = self.getCurrentL2Representation(L2Column0)
+    L2RepresentationA1 = self.getCurrentL2Representation(L2Column1)
```
Comment (subutai) [inline tests/layers/laminar_network_test.py]: Might want to ensure these representations have sufficient number of bits ON. I think the tests below could pass even if the L2 region was doing nothing and outputting all zeros - ??

File: htmresearch/frameworks/layers/laminar_network.py [hunk 3]
```diff
@@ -162,19 +162,22 @@ def createMultipleL4L2Columns(network, networkConfig):
       }
     }
   """
+  # FIXME: for now, each L2 column should have a different seed
```
Comment (subutai) [inline htmresearch/frameworks/layers/laminar_network.py]: Can you use "TODO:" instead of "FIXME" here and elsewhere?  Pycharm has a "TODO" tool 😄

**Comment by subutai** [comment]
@natoromano Looks good - left a few comments. Could you also add an initialization test where you set every parameter to a non-default value and check that the ColumnPooler object was actually constructed appropriately?

**Comment by natoromano** [comment]
OK, I will incorporate your comments.

**Comment by natoromano** [comment]
@subutai done. I had to make a few changes to the ColumnPooler region to enable custom proximal learning parameter to be passed (since there is a distinction between `synPermProximalInc` which refers to proximal learning and `permanenceIncrement` which is used for lateral learning, being passed to the member TM).

**Comment by subutai** [comment]
👍

**Pull request merged**